spec/features/text-input.rb

```diff
@@ -0,0 +1,134 @@
+require 'rubygems'
+require 'gosu'
+
+class TextField < Gosu::TextInput
+  # Some constants that define our appearance.
+  INACTIVE_COLOR  = 0xcc666666
+  ACTIVE_COLOR    = 0xccff6666
+  SELECTION_COLOR = 0xcc0000ff
+  CARET_COLOR     = 0xffffffff
+  PADDING = 5
+
+  attr_reader :x, :y
+
+  def initialize(window, font, x, y)
+    # TextInput's constructor doesn't expect any arguments.
+    super()
+
+    @window, @font, @x, @y = window, font, x, y
+
+    # Start with a self-explanatory text in each field.
+    self.text = "Click to change text"
+  end
+
+  # Example filter method. You can truncate the text to employ a length limit (watch out
+  # with Ruby 1.8 and UTF-8!), limit the text to certain characters etc.
+  def filter text
+    text.upcase
+  end
+
+  def draw
+    # Depending on whether this is the currently selected input or not, change the
+    # background's color.
+    if @window.text_input == self then
+      background_color = ACTIVE_COLOR
+    else
+      background_color = INACTIVE_COLOR
+    end
+    @window.draw_quad(x - PADDING,         y - PADDING,          background_color,
+                      x + width + PADDING, y - PADDING,          background_color,
```

Comment:
Line is too long. [82/80]<br>Align the parameters of a method call if they span more than one line.<br>Unnecessary spacing detected.